cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetectorUtils.java

```diff
@@ -28,10 +38,42 @@
   public static final String ANOMALY_DETECTION_TIME_MS_OBJECT_CONFIG = "anomaly.detection.time.ms.object";
   public static final long MAX_METADATA_WAIT_MS = 60000L;
   public static final Anomaly SHUTDOWN_ANOMALY = new BrokerFailures();
+  public static final Random RANDOM = new Random();
 
   private AnomalyDetectorUtils() {
   }
 
+  /**
+   * Create a Kafka consumer for retrieving reported Maintenance plans.
+   * The consumer uses {@link String} for keys and {@link MaintenancePlan} for values.
+   *
+   * @param configs The configurations for Cruise Control.
+   * @return A new Kafka consumer
+   */
+  public static Consumer<String, MaintenancePlan> createMaintenanceEventConsumer(Map<String, ?> configs, String groupIdPrefix) {
+    // Get bootstrap servers
+    String bootstrapServers = configs.get(MonitorConfig.BOOTSTRAP_SERVERS_CONFIG).toString();
+    // Trim the brackets in List's String representation.
+    if (bootstrapServers.length() > 2) {
+      bootstrapServers = bootstrapServers.substring(1, bootstrapServers.length() - 1);
+    }
+
+    // Create consumer
+    long randomToken = RANDOM.nextLong();
+    Properties consumerProps = new Properties();
+    consumerProps.putAll(configs);
+    consumerProps.setProperty(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+    consumerProps.setProperty(ConsumerConfig.GROUP_ID_CONFIG, groupIdPrefix + randomToken);
+    consumerProps.setProperty(ConsumerConfig.CLIENT_ID_CONFIG, groupIdPrefix + "-consumer-" + randomToken);
```

Comment:
Randomly generating the group id makes it impossible to track offset commits using lagmontor.  Randomly generating the client id breaks quotas based on client ids.  Also each time there is a new client id a new set of metrics are generated on both the broker and the client itself.  Since the brokers are probably longer lived than the clients this results is accumulated garbage.  Open source kafka does periodically clean it out but for some reason we've opted to keep them around in memory forever.  Recommend never having random numbers be part of the group or client ids.

Reply:
TL;DR Dropping group ids, but keeping random client ids.
_--_
> Randomly generating the group id makes it impossible to track offset commits using lagmontor

This consumer uses neither the group management functionality by using `subscribe(topic)` nor the Kafka-based offset management strategy. Hence, the `group.id` config is irrelevant. (Added a clarification to JavaDoc and dropped this setting).

> Randomly generating the client id ...

I see your point, but making this config deterministic is tricky:

If multiple CC instances point to the same cluster, which is a valid usage today (i.e. one active and multiple passive CC instances, for which all self-healing is disabled and `ReadOnlySampleStore` is used), then this will result in duplicate sensors for different clients. I will also share example internal issues caused by using the same client id for different consumers.

Reply:
There's usually some i00X number used to identify different instances, this can be part of the client id.  Other people can have other solutions to this problem.  Host name?

So CC never commits offsets into Kafka?  The offset management and the consumer group coordination can be used independently from each other.

Reply:
The previous code review was based on emitting the output in log messages.  I guess it's fine, but it seems strange because it's already a List it's possible to do the following in single line.
```java
  list.stream().collect(Collectors.joining( "," ) )
```

Reply:
The i00X string can be added to different cruise control instances.  For those outside of LinkedIn their deployment system may have something like this or they can add hostname or something else.

Reply:
> There's usually some i00X number used to identify different instances, this can be part of the client id. Other people can have other solutions to this problem. Host name?

> The i00X string can be added to different cruise control instances. For those outside of LinkedIn their deployment system may have something like this or they can add hostname or something else.

Even if a local (i.e. dev) deployment has a `i00X` string, I suspect that this would not suffice to distinguish clients for multiple local deployments pointing to _randomly configured_ (i.e. different) destination clusters (e.g. for testing).
If possible, I would prefer a solution that is applicable to both internal and open source. I believe there are at least 3 pieces that should go into the client id: (1) source host, (2) destination cluster, and (3) whether CC is the active instance or a passive one (and if passive, which specific passive instance).

Presumably, this change is not directly relevant to this PR other than continue using the existing approach. I created an issue about this discussion and propose to address this on a later patch: https://github.com/linkedin/cruise-control/issues/1337

> So CC never commits offsets into Kafka? The offset management and the consumer group coordination can be used independently from each other.

That is correct.